airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -159,12 +162,100 @@ def get_start_timestamp(self, stream_state) -> int:
 
         return start_point
 
+class IncrementalStripeStreamWithUpdates(IncrementalStripeStream):
+    """
+        This is a base class for incremental streams that support updates using the events API
+        In first sync it will not get any updates as the records are already updated
+        After first sync it will get all updates starting from date of the last sync
+    """
+    event_types = None
+    update_field = "event_created"
+    state_lastSync_key = "lastSync"
+    state_completed_key = "completed"
+    completed = False
+    def lookahead(self, iterable):
+        """Pass through all values from the given iterable, to check
+        if there are more values to come after the current one,
+        or if it is the last value. Helps to define in the last state the completed flag. 
+        """
+        # Get an iterator and pull the first value.
+        it = iter(iterable)
+        last = next(it)
+        # Run the iterator to exhaustion (starting from the second value).
+        for val in it:
+            yield last
+            last = val
+        self.completed = True
+        yield last
+    def shouldFetchFromOriginalResource (self,stream_state):
+        durationInDaysFromLastSync = 0
+        hasState = bool(stream_state)
+        self.completed = stream_state.get(self.state_completed_key) or False
+        if hasState and self.completed:
+            then = datetime.fromtimestamp(stream_state.get(self.state_lastSync_key)) 
+            now  = datetime.utcnow()
+            duration = now - then
+            durationInDaysFromLastSync = duration.days
```

Comment:
@ssbeefeater i think duration.days will round down to the nearest day. so in case the duration is between 30 and 31 days it will still return 30. In that case we may need to have the check for >= 30, rather than >30